chain/walker_test.go

```diff
@@ -57,22 +57,18 @@ func TestWalker(t *testing.T) {
 	strg, err := storage.NewDatabaseFromDB(ctx, db, "public")
 	require.NoError(t, err, "NewDatabaseFromDB")
 
-	taskAPI, err := NewDataSource(nodeAPI)
+	logging.SetAllLoggers(logging.LevelInfo)
+	taskAPI, err := datasource.NewDataSource(nodeAPI)
 	require.NoError(t, err)
-	tsIndexer, err := NewTipSetIndexer(taskAPI, strg, builtin.EpochDurationSeconds*time.Second, t.Name(), []string{BlocksTask})
-	require.NoError(t, err, "NewTipSetIndexer")
+	im, err := indexer.NewManager(taskAPI, strg, t.Name(), []string{indexer.BlocksTask}, indexer.WithWindow(builtin.EpochDurationSeconds*time.Second))
+	require.NoError(t, err, "NewManager")
 	t.Logf("initializing indexer")
-	idx := NewWalker(tsIndexer, nodeAPI, 0, int64(beforeHead.Height()))
+	idx := NewWalker(im, nodeAPI, 0, int64(head.Height()), false)
 
 	t.Logf("indexing chain")
 	err = idx.WalkChain(ctx, nodeAPI, head)
 	require.NoError(t, err, "WalkChain")
 
-	// TODO NewTipSetIndexer runs its processors in their own go routines (started when TipSet() is called)
```

Comment:
`WalkChain returns onces it's processed and persisted eveything, allow this to be removed.